server/src/Irony.cpp

```diff
@@ -258,17 +259,30 @@ void Irony::complete(const std::string &file,
       unsigned annotationStart = 0;
       bool typedTextSet = false;
 
-      if (availability == CXAvailability_NotAccessible ||
-          availability == CXAvailability_NotAvailable) {
-        continue;
-      }
-
       typedtext.clear();
       brief.clear();
       resultType.clear();
       prototype.clear();
       postCompCar.clear();
       postCompCdr.clear();
+      inaccess.clear();
+
+      // progress even if candidate is `CXAvailability_NotAccessible' due to the
+      // current Clang bug which evaluates a protected member to be inaccessible
```

Comment:
Please reference the bug here, so that one can check it when reading this code and maybe update the code once the bug is fixed.